x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/core/ml/config_index_mappings.json

```diff
@@ -43,6 +43,9 @@
               "maximum_number_trees" : {
```

Comment:
You can remove the old mapping can't you? If a new index is being created we don't need it and if the index is an old one then the mapping is already present?

Reply:
It makes sense to keep it around for the possibility that a job using the old field name is cloned into a new index.

Reply:
Won't the cloned job go through the process of renaming the field automatically as it is parsed and written out to xcontent. 

I think what is bothering me is that we can't comment on the mapping json `// remove this at 9.0.0` and no doubt it will be forgotten about. If it can go now I'd rather we get rid of it. 

Can you at least try removing the old mapping and see if the upgrade tests pass please

Reply:
That's a good point. There is a test failing but I can fix the test. I'll give it a go.

Reply:
Done